feat(curriculum): add linked list workshop (#61424)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-linked-list-class/68865003446e2f5d97ff2813.md

---
id: 68865003446e2f5d97ff2813
title: Step 12
challengeType: 20
dashedName: step-12
---

# --description--

> Still inside the `else` block, create a `while` loop that checks that `current_node.next` is not `None`, then add the `pass` keyword inside it.

Reviewer: We should combine this step with the next one.

```suggestion
Still inside the `else` block, create a `while` loop that checks that `current_node.next` is not `None`.

Inside the `while` loop, assign `current_node.next` to `current_node`.
```

# --hints--

You should create a `while` loop that check whether `current_node.next` is not `None`. Remeber to add the `pass` keyword inside the loop.

```js

> ({ test: () => runPython(`
> _cond = _Node(_code).find_class("LinkedList").find_function("add").find_ifs()[0].find_bodies()[1].find_whiles()[0]
> assert _cond.is_equivalent("while current_node.next is not None:\\n pass")

Reviewer: ```suggestion
({ test: () => runPython(`
_cond = _Node(_code).find_class("LinkedList").find_function("add").find_ifs()[0].find_bodies()[1].find_whiles()[0].find_conditions()[0]
assert _cond.is_equivalent("current_node.next is not None")
```
